src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -104,6 +104,12 @@ void Push(StackValue value)
         {
             FatalCheck(_stackTop < _maxStack, VerifierError.StackOverflow);
 
+            if (HasPendingPrefix(Prefix.ReadOnly))
```

Comment:
It looks a bit odd to have this logic here. I think it would be more appropriate to have it in the two IL instructions produce the read-only values. (I will be duplicated - but that's fine.)